Jenkinsfile

```diff
@@ -70,103 +55,148 @@ pipeline {
+              }
+            }
           }
         }
       }
       post {
         success {
           script {
-            if (params.RELEASE_BUILD && !params.RELEASE_DRY_RUN) {
-              Map buildInfoServer = [
-                "project_id": service_server,
-                "build_id": env.DOCKER_TAG,
-                "release_id": params.RELEASE_NAME,
-                "git_tag": env.GIT_TAG
-              ]
-              createBuild(
-                buildInfoServer,
-                rmsSupportedEnvironment.production
-              )
+            currentBuild.description = "${env.DOCKER_TAG} & ${env.SECONDARY_DOCKER_TAG}"
+            if (isHotfix) {
+              env.GIT_TAG = releaseTag.create(env.HOTFIX_NAME)
+            } else if (params.RELEASE_BUILD) {
+              env.GIT_TAG = releaseTag.getFormattedTag(params.RELEASE_NAME)
+              if (releaseTag.doesReleaseTagExist(params.RELEASE_NAME)) {
+                echo "REBUILD: Tag ${env.GIT_TAG} already exists"
+                return
+              }
+              if (params.RELEASE_DRY_RUN) {
+                echo "DRY RUN: Would have created ${env.GIT_TAG} and pushed it to the repo"
+                return
+              }
+              releaseTag.create(params.RELEASE_NAME)
             }
           }
         }
       }
     }
-    stage('Dockerize Secondary') {
+    stage('Publish') {
       when {
-        not { expression { isPr } }
+        allOf {
+          not { expression { isPr } }
+          not { expression { return params.RELEASE_BUILD && params.RELEASE_DRY_RUN } }
+        }
       }
-      steps {
-        script {
-          lastStage = env.STAGE_NAME
-          // Here's where we're getting the secondary artifact (named
-          // via env.STORE_PG_ARTIFACT) out for dockerizing, per the
-          // comment above.
-          if (params.RELEASE_BUILD) {
-            env.REGISTRY_PUSH = (params.RELEASE_DRY_RUN) ? 'none' : 'all'
-            env.SECONDARY_DOCKER_TAG = dockerize_secondary.docker_build_specify_tag_and_push(params.RELEASE_NAME, sbtbuild.getDockerPath(project_wd_secondary), env.STORE_PG_ARTIFACT, env.REGISTRY_PUSH)
-          } else {
-            env.REGISTRY_PUSH = 'internal'
-            env.SECONDARY_DOCKER_TAG = dockerize_secondary.docker_build('STAGING', env.GIT_COMMIT, sbtbuild.getDockerPath(project_wd_secondary), env.STORE_PG_ARTIFACT, env.REGISTRY_PUSH)
+      parallel {
+        stage('Publish - SoQL Server') {
+          steps {
+            script {
+              lastStage = env.STAGE_NAME
+              // Passing [] to overrideEnvironments will not override the environments, so the docker image will be published to all environments
+              Set environments = (params.RELEASE_BUILD) ? [] : ['internal']
```

Comment:
This solution makes the code DRY, and doesn't duplicate the full list of environments here.